src/api/auth.ts

```diff
@@ -135,221 +148,206 @@ function authenticatedUser(name: string, groups: string[] = []): any {
+    let hash = generateHash(password, config.secret);
+    let user = {
+      name: name,
+      password: hash,
+      fullName: fullName,
+      email: email
+    };
+    let index = auth.users.findIndex(u => u.name === name);
+    if (index === -1) {
+      auth.users.push(user);
+      resolve(auth);
+    } else {
+      reject('User with the same username allready exists.');
+    }
+  });
 }
 
-export function newOrganization(req: express.Request, res: express.Response):
-  express.Response | void {
-    let name = req.body.organization;
-    let username = req.body.username;
+export function newOrganization(organization: string, username: string): Promise<string> {
+  return new Promise((resolve, reject) => {
     let auth = getAuth();
-    let organization = {
-      name: name,
+    let org = {
+      name: organization,
       teams: [{ name: 'developers', members: [username] }],
       members: [{ name: username, role: 'owner' }],
       packages: []
     };
-    let index = auth.organizations.findIndex(org => org.name === name);
+    let index = auth.organizations.findIndex(org => org.name === organization);
     if (index === -1) {
-      auth.organizations.push(organization);
-      writeJsonFile(getAuthPath(), auth).then(
-        () => res.status(200).json({ success: 'Organization successfully added.' }));
+      auth.organizations.push(org);
+      resolve(auth);
     } else {
-      return res.status(200).json({ warning: 'Organization with the same name allready exists.' });
+      reject('Organization with the same name allready exists.');
     }
+  });
 }
 
-export function newTeam(req: express.Request, res: express.Response): express.Response | void {
-  let teamName = req.body.team;
-  let { organization, username } = req.body;
-  let auth = getAuth();
-  let team = {
-    name: teamName,
-    members: [ username ]
-  };
-  let orgIndex = auth.organizations.findIndex(org => org.name === organization);
-  let teamIndex = auth.organizations[orgIndex].teams.findIndex(t => t.name === teamName);
+export function newTeam(data: OrganizationTeamData): Promise<string> {
+  return new Promise((resolve, reject) => {
+    let { username, team, organization } = data;
+    let auth = getAuth();
+    let teamObject = {
+      name: team,
+      members: [ username ]
+    };
+    let orgIndex = auth.organizations.findIndex(org => org.name === organization);
+    let teamIndex = auth.organizations[orgIndex].teams.findIndex(t => t.name === team);
 
-  if (teamIndex === -1) {
-    auth.organizations[orgIndex].teams.push(team);
-    writeJsonFile(getAuthPath(), auth).then(() => {
-      res.status(200).json({ success: 'Team successfully added.' });
-    });
-  } else {
-    return res.status(200).json({ warning: 'Team with the same name allready exists.' });
-  }
+    if (teamIndex === -1) {
+      auth.organizations[orgIndex].teams.push(teamObject);
+      resolve(auth);
+    } else {
+      reject('Team with the same name allready exists.');
```

Comment:
What do you think about putting this text in a lang file instead of hardcoding it? Maybe it would give us more flexibility...